services/frontend/src/App.js

```diff
@@ -1,16 +1,40 @@
 import WeatherForecast from "./components/WeatherForecast";
 import SearchBar from "./components/SearchBar";
+import { useState } from "react";
 
 export const App = () => {
+  const defaultCity = {
+    city: "San Francisco",
+    state: "CA",
+  };
+
+  const [mainCity, setMainCity] = useState(defaultCity);
+  const [suggestedCities, setSuggestedCities] = useState([
+    { city: "Chicago", state: "IL" },
+    { city: "Oakland", state: "CA" },
+    { city: "New York", state: "NY" },
+  ]);
+
+  const updateMainCity = (newLocation) => {
+    if (
+      newLocation.city !== mainCity.city &&
+      newLocation.state !== mainCity.state
+    ) {
+      // Prepend prev. mainCity to top of suggested cities array
+      setSuggestedCities([mainCity, ...suggestedCities.slice(1, 3)]);
+    }
+    // React is smart enough to determine if the new content differs from the old.
+    // So we can do this outside the if statement.
+    setMainCity({ city: newLocation.city, state: newLocation.state });
+  };
+
   return (
     <div className="flex flex-col">
-      <SearchBar />
-      <WeatherForecast
-        locationData={{ city: "San Francisco", state: "CA" }}
-        defaultExpanded={true}
-      />
-      <WeatherForecast locationData={{ city: "Chicago", state: "IL" }} />
-      <WeatherForecast locationData={{ city: "Oakland", state: "CA" }} />
+      <SearchBar handleUserSearch={updateMainCity} />
```

Comment:
Notice that `mainCity` is a stateful property of the `App` class. But the action to update this property actually happens in the child `SearchBar` class. We handle this by declaring a method to update the state inside the `App` class and then pass that method down to the child component. `SearchBar` can then call that function to affect the state of its parent.

This is called "lifting up state". It's easy to do when components aren't deeply nested, but can get complicated quickly if the app gets big. 

Alternatives include the 3rd party library [Redux](https://react-redux.js.org/), which is very powerful but requires a *lot* of boilerplate, or React's new-ish [context API](https://reactjs.org/docs/context.html).